tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,368 @@
+#
+# Variable widths or heights in a grid
+# ------------------------------------
+#
+# Both `~.pyplot.subplots` and `~.pyplot.subplot_mosaic` allow the rows
+# in the grid to be different heights, and the columns to be different
+# widths using the *gridspec_kw* keyword argument.
+# Spacing parameters accepted by :class:`~matplotlib.gridspec.GridSpec`
+# can be passed to `~matplotlib.pyplot.subplots` and
+# `~matplotlib.pyplot.subplot_mosaic`:
+
+gs_kw = dict(width_ratios=[1.4, 1], height_ratios=[1, 2])
+fig, axd = plt.subplot_mosaic([['upleft', 'right'],
+                               ['loleft', 'right']],
+                              gridspec_kw=gs_kw, figsize=(4.5, 3.5),
+                              constrained_layout=True)
+for k in axd.keys():
+    annotate_axes(axd[k], f'axd["{k}"]', fontsize=14)
+fig.suptitle('plt.subplot_mosaic()')
+
+############################################################################
+# Nested Axes layouts
+# -------------------
+#
+# Sometimes it is helpful to have two or more grids of Axes that
+# may not need to be related to one another.  The most simple way to
+# accomplish this is to use `.Figure.subfigures`.  Note that the alignement
+# of the subfigure layouts is independent with the Axes spines in each
+# subfigure having independent positions.  See below for a more verbose
+# way to acheive the same effect with `~.gridspec.GridSpecFromSubplotSpec`.
+
+fig = plt.figure(constrained_layout=True)
+subfigs = fig.subfigures(1, 2, wspace=0.07, width_ratios=[1.5, 1.])
+axs0 = subfigs[0].subplots(2, 2)
+subfigs[0].set_facecolor('0.9')
+subfigs[0].suptitle('subfigs[0]\nLeft side')
+subfigs[0].supxlabel('xlabel for subfigs[0]')
+
+axs1 = subfigs[1].subplots(3, 1)
+subfigs[1].suptitle('subfigs[1]')
+subfigs[1].supylabel('ylabel for subfigs[1]')
+
+############################################################################
+# It is also possible to nest Axes using `~.pyplot.subplot_mosaic` using
+# nested lists.  This method does not use subfigures, like above, so lacks
+# the ability to add per-subfigure ``suptitle`` and ``supxlabel``, etc.
+# Rather it is a conveneince wrapper around the `~.SubplotSpec.subgridspec`
+# method described below.
+
+inner = [['innerA'],
+         ['innerB']]
+outer = [['upleft',  inner],
+          ['lowleft', 'lowright']]
+
+fig, axd = plt.subplot_mosaic(outer, constrained_layout=True)
+for k in axd.keys():
+    annotate_axes(axd[k], f'axd["{k}"]')
+
+############################################################################
+# Low-level and advanced grid methods
+# ===================================
+#
+# Internally, the arrangement of a grid of Axes is controlled by creating
+# instances of `~.GridSpec` and `~.SubplotSpec`. GridSpec defines a
+# (possibly non-uniform) grid of cells. Indexing into the GridSpec returns
+# a SubplotSpec that covers one or more grid cells, and can be used to
+# specify the location of an Axes or "subplot".
```

Comment:
Might be misinterpreted as "Axes or alternatively subplot" instead of "Axes, also called subplot".

I'm somewhat undecided how we should handle "subplot". Conceptually, it's "Axes located in a grid", but that technical distinction is irrelevant to 99% of our users, so I'm hesitant if we want to explain that. Since both Axes and "subplot" are prominent terms in our API we are also not able to just use one.

Probably the right way is to do whatever description we want in the note at the top (see the other comment). Here, we could just leave "subplot" out:

```suggestion
# specify the location of an Axes.
```
This doesn't remove relevant information. I'd also refrain from removing Axes and using "specify the location of a subplot". If we started that, we'd have to use subplot in very many places throught the tutorial; and it doesn't help that all our `ax*` variables are in fract Subplots rather than generic Axes, so calling it the "the subplot `ax`" would be even worse IMHO.

Reply:
I agree - I added this in response to another of your comments ;-)  But yeah, getting this out of the way at the beginning is better.